src/charm.py

```diff
@@ -94,7 +94,9 @@ def _patch_config(self, current_config):
                 if env in os.environ:
                     config[proxy] = os.environ[env]
         config["use-proxy"] = set(proxy_settings.values()) & set(config)
-        config["mirror-list"] = current_config["mirror-list"].splitlines()
+        config["mirror-list"] = self._validate_mirror_list(
+            current_config["mirror-list"]
+        )
         return config
 
     def _on_config_changed(self, _):
```

Comment:
**suggestion:** 

I saw a `try`, `except` and `else` python operators in this function. Please give a description about this function in doc-string

Reply:
Done